src/main.rs

```diff
@@ -60,25 +191,61 @@ fn parse_args() -> (String, String) {
     }
     let path = &args[1];
     let can_interface = &args[2];
+    let encode_en = args.len() >= 4 && args[3] == "encode_en";
 
     println!("Siren URL: {}", path);
     println!("Can interface: {}", can_interface);
 
-    (String::from(path), String::from(can_interface))
+    (String::from(path), String::from(can_interface), encode_en)
 }
 
 /**
  * Main Function
  * Configures the can network, retrieves the client based on the command line arguments,
  * connects the client and then reads the can socket from specified interface.
- * Sample Call for Mqtt "/home/ner/Desktop/Calypso/target/release/calypso mqtt localhost:1883 &"
  *
- * 3rd argument: if a can interface is passed the program will use it instead of the default can0
- * Ex: "/home/ner/Desktop/Calypso/target/release/calypso mqtt localhost:1883 can0 &"
+ * Args:
+ * 1 -> IP address:port of Siren mqtt server (ex. localhost:1883)
+ * 2 -> Socketcan interface name (ex. vcan0)
+ * 3 -> encode_en or blank, whether to enable encoding
  */
 fn main() {
-    let (path, can_interface) = parse_args();
-    let mut client = MqttClient::new();
-    client.connect(&path);
-    read_can(client, &can_interface);
+    let (path, can_interface, encoding) = parse_args();
+    let can_handle = match read_can(&path, &can_interface) {
+        Ok(handle) => handle,
+        Err(err) => {
+            println!("Decoder Errored! {}", err);
+            process::exit(1);
+        }
+    };
+
+    // use a arc for mutlithread, and a rwlock to enforce one writer
+    if encoding {
+        let send_map: Arc<RwLock<HashMap<u32, EncodeData>>> = Arc::new(RwLock::new(HashMap::new()));
+
+        let siren_handle = match read_siren(&path, Arc::clone(&send_map)) {
+            Ok(handle) => handle,
+            Err(err) => {
+                println!("Encoder Errored! {}", err);
+                process::exit(1);
```

Comment:
does this process exit kill calypso? if so i dont think we want to do that, calypso read should still work if calypso write fails for some reaosn